tests/Resources/ti.ui.tableview.test.js

```diff
@@ -1469,4 +1467,246 @@ describe('Titanium.UI.TableView', function () {
+			} catch (e) {
+				return finish(e);
+			}
+			finish();
+		});
+
+		win.add(tableView);
+		win.open();
+	});
+
+	it('TableViewRow internal icons', finish => {
+		win = Ti.UI.createWindow({ backgroundColor: 'gray' });
+
+		const tableView = Ti.UI.createTableView({
+			backgroundColor: 'white'
+		});
+
+		tableView.setData([
+			{ hasCheck: true },
+			{ hasChild: true },
+			{ hasDetail: true }
+		]);
+
+		win.addEventListener('postlayout', () => {
+			try {
+
+				// TableView should display rows of internal icons.
+				should(tableView).matchImage('snapshots/tableViewRow_icons.png');
+			} catch (e) {
+				return finish(e);
+			}
+			finish();
+		});
+
+		win.add(tableView);
+		win.open();
+	});
+
+	it('TableViewRow borderRadius', finish => {
+		win = Ti.UI.createWindow({ backgroundColor: 'gray' });
+
+		const tableView = Ti.UI.createTableView({
+			backgroundColor: 'white'
+		});
+
+		tableView.setData([
+			{
+				backgroundColor: 'blue',
+				height: 64,
+				borderRadius: 16
+			}
+		]);
+
+		win.addEventListener('postlayout', () => {
+			try {
+
+				// TableView should display rows of internal icons.
+				should(tableView).matchImage('snapshots/tableViewRow_borderRadius.png');
+			} catch (e) {
+				return finish(e);
+			}
+			finish();
+		});
+
+		win.add(tableView);
+		win.open();
+	});
+
+	it('TableViewRow default title & image', finish => {
+		win = Ti.UI.createWindow({ backgroundColor: 'gray' });
+
+		const tableView = Ti.UI.createTableView({
+			backgroundColor: 'white'
+		});
+
+		tableView.setData([
+			{
+				title: 'Default Title',
+				image: 'Logo.png'
+			}
+		]);
+
+		win.addEventListener('postlayout', () => {
+			try {
+
+				// TableView should display rows of internal icons.
+				should(tableView).matchImage('snapshots/tableViewRow_default.png');
+			} catch (e) {
+				return finish(e);
+			}
+			finish();
+		});
+
+		win.add(tableView);
+		win.open();
+	});
+
+	it('TableView headerTitle & footerTitle', finish => {
+		win = Ti.UI.createWindow({ backgroundColor: 'gray' });
+
+		const tableView = Ti.UI.createTableView({
+			headerTitle: 'Header',
+			footerTitle: 'Footer',
+			backgroundColor: 'white'
+		});
+
+		tableView.setData([ { title: 'Row' } ]);
+
+		win.addEventListener('postlayout', () => {
+			try {
+
+				// TableView should display rows of internal icons.
+				should(tableView).matchImage('snapshots/tableView_headerTitle_footerTitle.png');
+			} catch (e) {
+				return finish(e);
+			}
+			finish();
+		});
+
+		win.add(tableView);
+		win.open();
+	});
+
+	it('TableView headerView & footerView', finish => {
+		win = Ti.UI.createWindow({ backgroundColor: 'gray' });
+
+		const headerView = Ti.UI.createView({
+			backgroundColor: 'red'
+		});
+		const footerView = Ti.UI.createView({
+			backgroundColor: 'blue'
+		});
+		const tableView = Ti.UI.createTableView({
+			headerView,
+			footerView,
+			backgroundColor: 'white'
+		});
+
+		tableView.setData([{
+			title: 'Row'
+		}]);
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.ui.tableview.test.js;
  Line: 1659;
-->

- :warning: tests/Resources/ti.ui.tableview.test.js line 1659 – A space is required before ']'. (array-bracket-spacing)